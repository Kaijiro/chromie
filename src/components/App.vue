<template>
  This is my Application component !
</template>

<script lang="ts">
import {Vue} from 'vue-class-component';

export default class App extends Vue {
}
</script>
