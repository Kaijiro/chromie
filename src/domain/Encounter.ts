export default class Encounter {
    private fighters: Array<string> = [];

    countFighters(): number {
        return this.fighters.length;
    }

    addFighter(myFighter: string) {
        this.fighters.push(myFighter);
    }
}
