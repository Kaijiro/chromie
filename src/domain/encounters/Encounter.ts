export default class Encounter {
    private readonly name: string;
    private fighters: Array<string> = [];

    private constructor(name: string) {
        this.name = name;
    }

    public static with(name: string): Encounter{
        return new Encounter(name);
    }

    countFighters(): number {
        return this.fighters.length;
    }

    addFighter(fighter: string) {
        this.fighters.push(fighter);
    }

    getName() {
        return this.name;
    }
}
