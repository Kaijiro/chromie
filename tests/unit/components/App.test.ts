import {shallowMount} from "@vue/test-utils";
import App from "../../../src/components/App.vue";
import Encounter from "../../../src/domain/encounters/Encounter";

describe("App component", () => {

    it(`should display the drag and drop component when no file has been dropped`, () => {
        const mockStore = {
            state: {
                encounters: []
            }
        };
        const component = shallowMount(App, {
            global: {
                mocks: {
                    $store: mockStore
                }
            }
        });

        // It's currently a bit dirty to access these private attributes but this is the only way right now
        expect(component["__app"]._container).toMatchSnapshot();
    });

    it(`should display the sidebar and the main area when encounters are available`, () => {
        const encounter = new Encounter("A boss");
        const mockStore = {
            state: {
                encounters: [encounter]
            }
        };
        const component = shallowMount(App, {
            global: {
                mocks: {
                    $store: mockStore
                }
            }
        });

        // It's currently a bit dirty to access these private attributes but this is the only way right now
        expect(component["__app"]._container).toMatchSnapshot();
    });
});
