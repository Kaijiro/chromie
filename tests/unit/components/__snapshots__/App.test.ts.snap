// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should display the drag and drop component when no file has been dropped 1`] = `
<div
  data-v-app=""
>
  <file-drag-and-drop-stub />
</div>
`;

exports[`App component should display the sidebar and the main area when encounters are available 1`] = `
<div
  data-v-app=""
>
  <div>
    <sidebar-stub />
  </div>
</div>
`;
